packages/orchestration/test/test-withdraw-reward.js

```diff
@@ -0,0 +1,195 @@
+// @ts-check
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+
+import { MsgWithdrawDelegatorRewardResponse } from '@agoric/cosmic-proto/cosmos/distribution/v1beta1/tx.js';
+import { MsgDelegateResponse } from '@agoric/cosmic-proto/cosmos/staking/v1beta1/tx.js';
+import { Any } from '@agoric/cosmic-proto/google/protobuf/any.js';
+import { makeScalarBigMapStore } from '@agoric/vat-data';
+import { encodeBase64 } from '@endo/base64';
+import { E, Far } from '@endo/far';
+import { prepareStakingAccountKit } from '../src/exos/stakingAccountKit.js';
+
+/**
+ * @import {ChainAccount} from '../src/types.js';
+ * @import { Coin } from '@agoric/cosmic-proto/cosmos/base/v1beta1/coin.js';
+ */
+
+const test = anyTest;
+
+const { Fail } = assert;
+
+const scenario1 = {
+  acct1: {
+    address: 'agoric1spy36ltduehs5dmszfrp792f0k2emcntrql3nx',
+  },
+  validator: { address: 'agoric1valoper234', addressEncoding: 'bech32' },
+  delegations: {
+    agoric1valoper234: { denom: 'uatom', amount: '200' },
+  },
+};
+
+const makeScenario = () => {
+  const txEncode = (response, toProtoMsg) => {
+    const protoMsg = toProtoMsg(response);
+    const any1 = Any.fromPartial(protoMsg);
+    const any2 = Any.fromPartial({ value: Any.encode(any1).finish() });
+    const ackStr = encodeBase64(Any.encode(any2).finish());
+    return ackStr;
+  };
+
+  /**
+   * @param {string} [addr]
+   * @param {Record<string, Coin>} [delegations]
+   */
+  const mockAccount = (addr = 'agoric1234', delegations = {}) => {
+    const calls = [];
+
+    const simulate = {
+      '/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward': m => {
+        console.log('simulate withdraw', m);
+        const rewards = Object.values(delegations).map(({ denom, amount }) => ({
+          denom,
+          amount: `${Number(amount) / 100}`,
+        }));
+        /** @type {MsgWithdrawDelegatorRewardResponse} */
+        const response = { amount: rewards };
+
+        return txEncode(
+          response,
+          MsgWithdrawDelegatorRewardResponse.toProtoMsg,
+        );
+      },
+
+      '/cosmos.staking.v1beta1.MsgDelegate': _m => {
+        const response = MsgDelegateResponse.fromPartial({});
+        return txEncode(response, MsgDelegateResponse.toProtoMsg);
+      },
+    };
+
+    /** @type {ChainAccount} */
+    const account = Far('MockAccount', {
+      getAccountAddress: () => addr,
+      executeEncodedTx: async msgs => {
+        assert.equal(msgs.length, 1);
+        const { typeUrl } = msgs[0];
+        const doMessage = simulate[typeUrl];
+        assert(doMessage, `unknown ${typeUrl}`);
+        await null;
+        calls.push({ msgs });
+        return doMessage(msgs[0]);
+      },
+      executeTx: () => Fail`mock`,
+      close: () => Fail`mock`,
+      deposit: () => Fail`mock`,
+      getPurse: () => Fail`mock`,
+      prepareTransfer: () => Fail`mock`,
+    });
+    return { account, calls };
+  };
+
+  const mockZCF = () => {
```

Comment:
Curious, there's nothing we could import here?

Reply:
maybe there is, but I'm not familiar with it.
In this case of build-vs-buy, it seemed easier to write a few lines than to get into the car to go shopping.

Reply:
I looked around a bit; I don't see anything in  `zoe/tools`.
The closest I found is `zoe/test/unitTests/zcf/setupZcfTest.js`, but that has a hard-coded contract.